controller/lib/core/sensors.cpp

```diff
@@ -141,15 +141,14 @@ void TVIntegrator::AddFlow(Time now, VolumetricFlow flow) {
   //
   //  - Measure time with better than millisecond granularity.
   if (!initialized_) {
-    last_flow_ = flow;
-    last_flow_measurement_time_ = now;
     initialized_ = true;
   } else if (Duration delta = now - last_flow_measurement_time_;
              delta >= VOLUME_INTEGRAL_INTERVAL) {
-    volume_ = volume_ + ml(delta.minutes() *
-                           (last_flow_.ml_per_min() + flow.ml_per_min()) / 2);
-    last_flow_measurement_time_ = now;
+    volume_ =
+        volume_ + ml(delta.minutes() * (last_flow_ + flow).ml_per_min() / 2.0f);
   }
+  last_flow_ = flow;
+  last_flow_measurement_time_ = now;
```

Comment:
Yeah, I don't think this is right.  You only want to update last_flow_ and last_flow_measurement_time_ if you went into one of the two branches above.

Reply:
indeed, I didn't see that compounded else statement!

Reply:
my unit test run with a constant sampling which is higher than VOLUME_INTEGRAL_INTERVAL...

Reply:
Given that we've written this wrong twice now...I wonder if we could fix the code so we can't have this bug.  I think that would mean getting rid of the `initialized_` variable.

One way to do it would be to set last_flow_=0 and last_flow_measurement_time_=now() when the TVIntegrator is constructed.  I sort of think that's OK?

Another way to do it would be to make TVIntegrator an std::optional inside of Sensors, then make TVIntegrator's constructor take the current flow.

Reply:
I'll add that to the long list of tests I am not satisfied with, and do something about it!

Reply:
> One way to do it would be to set last_flow_=0 and last_flow_measurement_time_=now() when the TVIntegrator is constructed. I sort of think that's OK?

I guess we could do that... will run my new tests and see what they give with that change